File: 0674. Longest Continuous Increasing Subsequence/0674.py
# time complexity: O(n)
# space complexity: O(1)
class Solution(object):
    # two-pointer (sliding window) approach
    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        max_len = 1        # max length so far
        left = 0           # start index of current window
        
        for right in range(1, len(nums)):
            # if sequence breaks, move left to right
            if nums[right] <= nums[right - 1]:
                left = right
            # window size is right‑left+1
            max_len = max(max_len, right - left + 1)
        
        return max_len

File: 0674. Longest Continuous Increasing Subsequence/0674.java
// time complexity: O(n)
// space complexity: O(1)
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        // handle edge case if needed (LeetCode guarantees nums.length >= 1)
        if (nums == null || nums.length == 0) return 0;

        int longest = 0;    // record of max length seen
        int currLen = 1;    // length of current increasing run

        // scan through array
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                currLen++;               // extend current run
            } else {
                longest = Math.max(longest, currLen);  // update best
                currLen = 1;            // reset for new run
            }
        }

        // final check in case the longest run goes to the end
        return Math.max(longest, currLen);
    }
}